crates/transaction-pool/src/metrics.rs

```diff
@@ -36,8 +36,18 @@ pub struct TxPoolMetrics {
     /// Total amount of memory used by the transactions in the blob sub-pool in bytes
     pub(crate) blob_pool_size_bytes: Gauge,
 
-    /// Number of all transactions of all sub-pools: pending + basefee + queued
+    /// Number of all transactions of all sub-pools: pending + basefee + queued + blob
     pub(crate) total_transactions: Gauge,
+    /// Number of all legacy transactions in the pool
+    pub(crate) total_legacy_transactions: Gauge,
+    /// Number of all EIP-2930 transactions in the pool
+    pub(crate) total_eip2930_transactions: Gauge,
+    /// Number of all EIP-1559 transactions in the pool
+    pub(crate) total_eip1559_transactions: Gauge,
+    /// Number of all EIP-4844 transactions in the pool
+    pub(crate) total_eip4844_transactions: Gauge,
+    /// Number of all EIP-7702 transactions in the pool
+    pub(crate) total_eip7702_transactions: Gauge,
```

Comment:
This wasn't mentioned in the original issue but I assume we wanted to track all tx types so added here.